src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/EntityLocation.java

```diff
@@ -0,0 +1,179 @@
+import com.minecolonies.api.colony.requestsystem.factory.IFactoryController;
+import com.minecolonies.api.colony.requestsystem.location.ILocation;
+import com.minecolonies.api.colony.requestsystem.location.ILocationFactory;
+import net.minecraft.entity.Entity;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.ref.WeakReference;
+import java.util.UUID;
+
+/**
+ * Location described by an Entity.
+ */
+public class EntityLocation implements ILocation
+{
+
+    @NotNull
+    private final UUID uuid;
+
+    @Nullable
+    private WeakReference<Entity> entity;
+
+    EntityLocation(@NotNull final UUID uuid)
+    {
+        this.uuid = uuid;
+        checkEntity();
+    }
+
+    private void checkEntity()
+    {
+        if (entity != null)
+        {
+            return;
+        }
+
+        entity = new WeakReference<>(FMLCommonHandler.instance().getMinecraftServerInstance().getEntityFromUuid(uuid));
+    }
+
+    /**
+     * Method to get the location in the dimension
+     *
+     * @return The location.
+     */
+    @NotNull
+    @Override
+    public BlockPos getInDimensionLocation()
+    {
+        checkEntity();
+        if (entity == null || entity.get() == null)
+        {
+            return BlockPos.ORIGIN;
+        }
+        else
+        {
+            return entity.get().getPosition();
+        }
+    }
+
+    /**
+     * Method to get the dimension of the location.
+     *
+     * @return The dimension of the location.
+     */
+    @Override
+    public int getDimension()
+    {
+        checkEntity();
+        if (entity == null || entity.get() == null)
+        {
+            return 0;
+        }
+        else
+        {
+            return entity.get().dimension;
+        }
+    }
+
+    /**
+     * Method to check if this location is reachable from the other.
+     *
+     * @param location The check if it is reachable from here.
+     * @return True when reachable, false when not.
+     */
+    @Override
+    public boolean isReachableFromLocation(@NotNull final ILocation location)
+    {
+        checkEntity();
+        return !(entity == null || entity.get() == null) && location.getDimension() == getDimension();
+    }
+
+    @SuppressWarnings("squid:S2972")
+    /**
+     * We have this class the way it is for a reason.
+     */
+    public static class Factory implements ILocationFactory<Entity, EntityLocation>
+    {
+
+        ////// --------------------------- NBTConstants --------------------------- \\\\\\
+        private static final String NBT_MSB = "Id_MSB";
+        private static final String NBT_LSB = "Id_LSB";
+        ////// --------------------------- NBTConstants --------------------------- \\\\\\
+
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:LeftCurlyBraceStartLineCheck' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)